### DISCUSSION TITLE
Displaying data

### DISCUSSION THREAD

**Discussion by rbruno1** [jspsych/jsPsych#1883] [Q&A] [answered]
Hi. I am trying to make my experiemtn display and record data in a clean way. this is the source code for one of my trials:
'''
const Tur_AV_trial_10 = {
        type:'video-keyboard-response',
       stimulus: [
         'mushroommuffin0j.mp4'
          ],
        post_trial_gap: 1000,
        choices: ['f', 'j'],
        data: {condition: 'AVTur', correct_response: 'j'},
        width:1000
}
'''
What do i need to add to this code and the "on_finish:" section in my Jspsych.init to get data on the participants accuracy and whether they had a correct response?

Thank you

**Comment by becky-gilbert** [comment, marked as answer]
Hi @rbruno1! To score each response as correct or incorrect, you can add an `on_finish` parameter to the trial that compares the correct response (which you've already stored in the trial data) with the participant's response (which is added to the trial data when the trial is finished). It would look like this:
```js
const Tur_AV_trial_10 = {
  // other parameters...
  on_finish: function(data) {
    data.correct = jsPsych.pluginAPI.compareKeys(data.response, data.correct_response);
  }
}
```
This code will add a `correct` property to the data for each trial. Its value will be `true` if the response matches the correct response, and `false` if not.

You can take a look at the [Manipulating Data section of the Simple Reaction Time tutorial](https://www.jspsych.org/tutorials/rt-task/#part-12-manipulating-data-during-the-experiment) for another example and explanation of scoring the response from each trial. And there's a section about [Data Aggregation](https://www.jspsych.org/tutorials/rt-task/#part-13-data-aggregation) to get a performance summary across trials, if you're interested in that too.

I hope that helps. Let us know if you have any other questions.

**Reply by rbruno1** [reply]
hi. It is telling me that data is not defined. I thought i defined it in this code: "data: {condition: 'AVTur', correct_response: 'j'},"

**Reply by becky-gilbert** [reply]
Sorry, just noticed an error in my response. The on_finish function automatically takes the trial data as an argument, which I forgot to include. So it should look like this:
```js
on_finish: function(data) {
  // ...
}
```

**Reply by rbruno1** [reply]
you are amazing.
